src/main/java/org/mastodon/mamut/feature/SpotGaussFilteredIntensityFeatureComputer.java

```diff
@@ -166,11 +214,54 @@ public void run()
 					}
 
 					final double variance = S / weightedSum;
-					output.means.get( iSource ).put( spot, weightedMean );
-					output.stds.get( iSource ).put( spot, Math.sqrt( variance ) );
+					output.means.get( iSource ).set( spot, weightedMean );
+					output.stds.get( iSource ).set( spot, Math.sqrt( variance ) );
+				}
+			}
+		}
+	}
+
+	public static final long nSpots( final IntFunction< Iterable< Spot > > index, final int numTimepoints )
+	{
+		long nSpots = 0l;
+		for ( int timepoint = 0; timepoint < numTimepoints; timepoint++ )
+		{
+			final Iterable< Spot > iterable = index.apply( timepoint );
+			for ( @SuppressWarnings( "unused" ) final Spot spot : iterable )
+				nSpots++;
+		}
+		return nSpots;
+	}
+
+	private static final class MyIndex implements IntFunction< Iterable< Spot > >
+	{
+
+		private final Map< Integer, Collection< Spot > > index;
+
+		public MyIndex( final GraphUpdate< Spot, Link > update )
+		{
+			this.index = new HashMap<>();
+			for ( final Spot spot : update.vertices( UpdateLocality.SELF ) )
+			{
+				final int timepoint = spot.getTimepoint();
+				Collection< Spot > iterable = index.get( Integer.valueOf( timepoint ) );
+				if ( null == iterable )
```

Comment:
Can be replaced with `computeIfAbsent`.
Integer.valueOf() is not needed.

Reply:
Noted.